File: accommodation_service/src/main/java/huy/project/accommodation_service/core/service/impl/LanguageService.java
package huy.project.accommodation_service.core.service.impl;

import huy.project.accommodation_service.core.domain.dto.request.CreateLanguageRequestDto;
import huy.project.accommodation_service.core.domain.entity.LanguageEntity;
import huy.project.accommodation_service.core.service.ILanguageService;
import huy.project.accommodation_service.core.usecase.CreateLanguageUseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LanguageService implements ILanguageService {
    private final CreateLanguageUseCase createLanguageUseCase;

    @Override
    public LanguageEntity createLanguage(CreateLanguageRequestDto req) {
        return createLanguageUseCase.createLanguage(req);
    }
}

File: location_service/LocationService.Core/Service/LanguageService.cs
namespace LocationService.Core.Service
{
    using System.Threading.Tasks;
    using LocationService.Core.Domain.Dto.Request;
    using LocationService.Core.Domain.Entity;
    using LocationService.Core.UseCase;

    public interface ILanguageService
    {
        Task<LanguageEntity> CreateLanguageAsync(CreateLanguageDto languageDto);
    }

    public class LanguageService : ILanguageService
    {
        private readonly ICreateLanguageUseCase _createLanguageUseCase;

        public LanguageService(ICreateLanguageUseCase createLanguageUseCase)
        {
            _createLanguageUseCase = createLanguageUseCase;
        }

        public async Task<LanguageEntity> CreateLanguageAsync(CreateLanguageDto languageDto)
        {
            return await _createLanguageUseCase.CreateLanguageAsync(languageDto);
        }
    }
}